README.md

```diff
@@ -1 +1,106 @@
+
+### Usage
+
+Create the following file in your module, and substitute the tagged version for the most recent `v` from this module e.g. `@v1`
+
+**.github/workflows/ci.yml**
+```yml
+name: CI
+
+on:
+  push:
+  pull_request:
+  workflow_dispatch:
+
+jobs:
+  ci:
+    uses: silverstripe/ghi-ci/.github/workflows/ci.yml@main
+```
+
+Set config specific to your needs via "inputs" defined under the `with:` key.  For instance, to disable PHP linting because you module does not does not yet have a `phpcs.xml.dist` file
+
+```yml
+jobs:
+  ci:
+    uses: silverstripe/github-actions-ci-cd/.github/workflows/ci.yml@main
+    with:
+      phplinting: false
+```
+
+#### Inputs:
+
+*Extra composer requirements*
+Require additional modules through composer for matrix entries. You do not need to quote the string for multiple requirements
+`composer_require_extra: silverstripe/widgets:^2 silverstripe/comments:^3`
+
+*Simple matrix*
+Create a smaller matrix with only the lowest supported PHP version lowest supported MySQL only instead of a full matrix of multiple PHP and database versions. Default is false, enable with:
+`simple_matrix: true`
+
+*Dyanmic matrix*
+Dynamically generate a matrix using feature detection. If disabled, jobs must be defined using the extra_jobs input. Default is true, disable with:
+`dynamic_matrix: false`
+
+*PHPUnit tests*
+Runs phpunit if config file is phpunit.xml or phpunit.xml.dist is available. Default is true, disable phpunit tests with:
+`phpunit: false`
+
+*PHP linting*
+Runs phpcs and phpstan if config files phpcs.xml.dist or phpstan.neon.dist are available. Default is true, disable with:
+`phplinting: false`
+
+*PHP coverage*
+Run codecov, does not require a configuration file. Default is false, though modules on the silverstripe account will always have this enabled, enable with:
+`phpcoverage: true`
+
+*End-to-end tests*
+Runs behat tests is behat.yml is available. Default is true, disable end-to-end tests with:
+`endtoend: false`
+
+*JS tests*
+Runs JS tests such as yarn lint, yarn test and yarn build diff if config file package.json is available. Default is true, disable JS tests with:
+`js: false`
+
+*Extra jobs*
+Define php version and/or db.  All inputs are false by default so you only need to include config for the input you want enabled. All inputs except `dynamic_matrix` and `simple_matrix` are available in this context. Use a multi-line yml string.
+```yml
+extra_jobs: |
+  - php: '8.0'
+    db: pgsql
+    phpunit: true
+    phpunit_suite: api
+  - php: '8.1'
+    endtoend: true
+    endtoend_suite: admin
+    endtoend_config: vendor/silverstripe/admin/behat.yml
+```
+
+*Extra jobs - additional inputs*
+Additional inputs that can be applied to individual jobs within the extra_jobs input
```

Comment:
They're technically inputs not options

Reply:
I would argue that saying "inputs" here is confusing because they're just a string _on_ an input... the fact that they're passed into some action later down the chain as inputs isn't useful information here. But it's a tiny detail not worth fighting over, I'm okay leaving this as is.